src/main/java/tech/sourced/siva/ByteBufferBackedInputStream.java

```diff
@@ -4,22 +4,38 @@
 import java.io.InputStream;
 import java.nio.ByteBuffer;
 
+/**
+ * An {@link InputStream} backed by a {@link ByteBuffer} to improve read operations.
+ */
 // http://nadeausoftware.com/articles/2008/02/java_tip_how_read_files_quickly
 class ByteBufferBackedInputStream extends InputStream {
 
     private final ByteBuffer buf;
 
+    /**
+     * Constructor.
```

Comment:
I'd put something more descriptive like `Constructs a new {@link ByteBufferBackedInputStream} backed by the given {@link ByteBuffer}.`